### ISSUE TITLE
cmake policy CMP0114

### ISSUE THREAD

**Issue by nickdesaulniers** [llvm/llvm-project#54944]
Doing a build of the 14.0.0 and 14.0.1 releases (I haven't tested older releases) produces a stream of warnings for me (for LLVM_ENABLE_RUNTIMES build):
```
CMake Warning (dev) at /usr/share/cmake/Modules/ExternalProject.cmake:2147 (message):
  Policy CMP0114 is not set: ExternalProject step targets fully adopt their
  steps.  Run "cmake --help-policy CMP0114" for policy details.  Use the
  cmake_policy command to set the policy and suppress this warning.

  ExternalProject target 'builtins' would depend on the targets for step(s)
  'clean' under policy CMP0114, but this is being left out for compatibility
  since the policy is not set.
Call Stack (most recent call first):
  cmake/modules/LLVMExternalProjectUtils.cmake:345 (ExternalProject_Add_StepTargets)
  runtimes/CMakeLists.txt:82 (llvm_ExternalProject_Add)
  runtimes/CMakeLists.txt:140 (builtin_default_target)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /usr/share/cmake/Modules/ExternalProject.cmake:2147 (message):
  Policy CMP0114 is not set: ExternalProject step targets fully adopt their
  steps.  Run "cmake --help-policy CMP0114" for policy details.  Use the
  cmake_policy command to set the policy and suppress this warning.

  ExternalProject target 'runtimes' would depend on the targets for step(s)
  'clean' under policy CMP0114, but this is being left out for compatibility
  since the policy is not set.
Call Stack (most recent call first):
  cmake/modules/LLVMExternalProjectUtils.cmake:345 (ExternalProject_Add_StepTargets)
  runtimes/CMakeLists.txt:232 (llvm_ExternalProject_Add)
  runtimes/CMakeLists.txt:375 (runtime_default_target)
This warning is for project developers.  Use -Wno-dev to suppress it.
```
I'm guessing we could do something like:
```diff
diff --git a/llvm/CMakeLists.txt b/llvm/CMakeLists.txt
index 5e786b712051..d21bcd0a2c3a 100644
--- a/llvm/CMakeLists.txt
+++ b/llvm/CMakeLists.txt
@@ -2,6 +2,11 @@
 
 cmake_minimum_required(VERSION 3.13.4)
 
+# CMP0114: ExternalProject has new behavior.
+# New in CMake 3.19: https://cmake.org/cmake/help/latest/policy/CMP0114.html
+if(POLICY CMP0114)
+  cmake_policy(SET CMP0114 OLD)
+endif()
 # CMP0116: Ninja generators transform `DEPFILE`s from `add_custom_command()`
 # New in CMake 3.20. https://cmake.org/cmake/help/latest/policy/CMP0116.html
 if(POLICY CMP0116)
```
but perhaps it's better to fix our cmake to upgrade to the new behavior, if possible?

**Comment by penzn** [comment]
Looks like the same warning reported in #50122

**Comment by nickdesaulniers** [comment]
https://reviews.llvm.org/D126923
